gammapy/modeling/models/spatial.py

```diff
@@ -1282,3 +1286,129 @@ def plot_interactive(self, ax=None, geom=None, **kwargs):
         if geom is None:
             geom = self.map.geom
         super().plot_interactive(ax=ax, geom=geom, **kwargs)
+
+
+class PiecewiseNormSpatialModel(SpatialModel):
+    """Piecewise spatial correction
+       with a free normalization at each fixed nodes.
+
+       For more information see :ref:`piecewise-norm-spectral`.
+
+    Parameters
+    ----------
+    coord : `gammapy.maps.MapCoord`
+        Flat coordinates list at which the model values are given (nodes).
+    norms : `~numpy.ndarray` or list of `Parameter`
+        Array with the initial norms of the model at energies ``energy``.
+        A normalisation parameters is created for each value.
+        Default is one at each node.
+    interp : str
+        Interpolation scaling in {"log", "lin"}. Default is "lin"
+    """
+
+    tag = ["PiecewiseNormSpatialModel", "piecewise-norm"]
+
+    def __init__(self, coords, norms=None, interp="lin", **kwargs):
+
+        self._coords = coords
+        self._interp = interp
+
+        if norms is None:
+            norms = np.ones(coords.shape)
+
+        if len(norms) != coords.shape[0]:
+            raise ValueError("dimension mismatch")
+
+        if len(norms) < 4:
+            raise ValueError("Input arrays must contain at least 4 elements")
+
+        if not isinstance(norms[0], Parameter):
+            parameters = Parameters(
+                [Parameter(f"norm_{k}", norm) for k, norm in enumerate(norms)]
+            )
+        else:
+            parameters = Parameters(norms)
+        self.default_parameters = parameters
+        super().__init__(**kwargs)
+
+    @property
+    def coords(self):
+        """Energy nodes"""
+        return self._coords
+
+    @property
+    def norms(self):
+        """Norm values"""
+        return u.Quantity(self.parameters.value)
+
+    @property
+    def is_energy_dependent(self):
+        keys = self.coords._data.keys()
+        return "energy" in keys or "energy_true" in keys
+
+    def evaluate(self, lon, lat, energy=None, **norms):
+        """Evaluate the model at given coordinates."""
+        scale = interpolation_scale(scale=self._interp)
+        v_nodes = scale(self.norms)
+        coords = [value.value for value in self.coords._data.values()]
+        # TODO: apply axes scaling in this loop
+        coords = list(zip(*coords))
+        if self.is_energy_dependent:
+            if energy is None:
+                raise ValueError("Missing nergy value for  energy-dependent model")
+            interpolated = griddata(
+                coords, v_nodes, (lon, lat, energy), method="linear"
+            )
+            scaled_norms = scale.inverse(interpolated)
+        else:
+            interp = CloughTocher2DInterpolator(coords, v_nodes)
```

Comment:
why the different interpolation scheme here?

Reply:
CloughTocher2DInterpolator is equivalent to use griddata with method="cubic",  by default I would rely on this because it has some nice properties (C1 smooth, curvature-minimization). I simplified the code but left a comment about this.